x/registry/types/who_is.go

```diff
@@ -2,214 +2,93 @@ package types
-	}
-
-	if an := m["app_name"]; an != "" {
-		return an, nil
+// UnmarshalDidDocument unmarshals the whois document into a DID document
+func (w *WhoIs) UnmarshalDidDocument() (*did.Document, error) {
+	doc := did.Document{}
+	err := doc.UnmarshalJSON(w.DidDocument)
+	if err != nil {
+		return nil, err
 	}
-	return "", fmt.Errorf("no app_name found")
+	return &doc, nil
 }
 
-// ApplicationDescription returns the description of the application
-func (w *WhoIs) ApplicationDescription() (string, error) {
-	if w.IsUser() {
-		return "", fmt.Errorf("not an application")
-	}
-	m := w.GetMetadata()
-	if m == nil {
-		return "", fmt.Errorf("no metadata found")
-	}
-
-	if an := m["app_description"]; an != "" {
-		return an, nil
+// CopyFromDidDocument copies the DID document into the WhoIs object if the DID document has the same DID owner
+func (w *WhoIs) CopyFromDidDocument(doc *did.Document) error {
+	if w.Owner != strings.TrimLeft(doc.ID.ID, "did:snr:") {
+		return fmt.Errorf("owner mismatch: %s != %s", w.Owner, doc.ID.ID)
 	}
-	return "", fmt.Errorf("no app_description found")
+	w.AddAlsoKnownAs(doc.AlsoKnownAs)
+	w.Controllers = doc.ControllersAsString()
+	w.Timestamp = time.Now().Unix()
+	w.IsActive = true
+	return nil
 }
 
-// ApplicationCategory returns the category of the application
-func (w *WhoIs) ApplicationCategory() (string, error) {
-	if w.IsUser() {
-		return "", fmt.Errorf("not an application")
-	}
-	m := w.GetMetadata()
-	if m == nil {
-		return "", fmt.Errorf("no metadata found")
-	}
-
-	if an := m["app_category"]; an != "" {
-		return an, nil
+// GetAlsoKnownAs returns the list of aliases of the whois as string array
+func (w *WhoIs) GetAlsoKnownAs() []string {
+	var aliases []string
+	for _, a := range w.Alias {
+		aliases = append(aliases, a.GetName())
 	}
-	return "", fmt.Errorf("no app_category found")
+	return aliases
 }
 
-// ApplicationUrl returns the website of the application
-func (w *WhoIs) ApplicationUrl() (string, error) {
-	if w.IsUser() {
-		return "", fmt.Errorf("not an application")
+// AddAlsoKnownAs adds an alias to the list of aliases of the whois
+func (w *WhoIs) AddAlsoKnownAs(as []string) error {
+	if w.Alias == nil {
+		w.Alias = make([]*Alias, 0)
 	}
-	m := w.GetMetadata()
-	if m == nil {
-		return "", fmt.Errorf("no metadata found")
-	}
-
-	if an := m["app_url"]; an != "" {
-		return an, nil
-	}
-	return "", fmt.Errorf("no app_description found")
-}
 
-// CredentialExcludeList returns a CredentialDescriptor array filled
-// with all the user's credentials
-func (w *WhoIs) CredentialExcludeList() []protocol.CredentialDescriptor {
-	credentialExcludeList := []protocol.CredentialDescriptor{}
-	for _, cred := range w.GetCredentials() {
-		descriptor := protocol.CredentialDescriptor{
-			Type:         protocol.PublicKeyCredentialType,
-			CredentialID: cred.ID,
+	for _, a := range as {
+		if !w.ContainsAlias(a) {
+			w.Alias = append(w.Alias, &Alias{
+				Name:      a,
+				IsForSale: false,
+				Amount:    10,
```

Comment:
This should be unset if the item is not for sell (0 or -1 maybe)

Reply:
True will do